akka-actor-testkit-typed/src/main/scala/akka/actor/testkit/typed/internal/ActorSystemStub.scala

```diff
@@ -53,8 +53,9 @@ import scala.concurrent._
     terminationPromise.trySuccess(Terminated(this)(null))
     terminationPromise.future
   }
-  override def whenTerminated: Future[akka.actor.typed.Terminated] = terminationPromise.future
-  override def getWhenTerminated: CompletionStage[Terminated] = FutureConverters.toJava(whenTerminated)
+  private val cachedWhenTerminatedFuture = terminationPromise.future.map(_ ⇒ Done)
```

Comment:
why not change terminationPromise to `Done`? do we need the `Terminated` elsewhere?

Reply:
The `terminate` method still uses `Terminated`?